src/shogun/base/SGObject.h

```diff
@@ -390,9 +375,38 @@ class CSGObject
 		put(Tag<T*>(name), value);
 	}
 
+	/** Typed setter for an enum mapped with a string, identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @param value value of the parameter
+	 */
+	template <class T>
+	void
+	put(const std::string& name, const std::string& value) noexcept(false)
```

Comment:
Just realising. We might at some point have string parameters....

Reply:
Ah but that's fine. We can just check if the parameter is in the map, and if not dispatch it with `CSGObject::put(const Tag<T>& _tag, const T& value)` inside this function.